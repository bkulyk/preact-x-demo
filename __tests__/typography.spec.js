import React from 'react';
import Typography from '@bitchin/typography';
import { shallow } from './enzyme';
import { it, expect } from './jest'

it('should render correctly', () => {
  expect(shallow((
    <Typography>
      some text
    </Typography>
  ))).toMatchSnapshot();
});

it('should render correctly', () => {
  expect(shallow((
    <Typography
      use="headline5"
      component="p"
    >
      some text
    </Typography>
  ))).toMatchSnapshot();
});

it('should render correctly', () => {
  expect(shallow((
    <Typography
      use="overline"
      component="div"
      className="custom"
      ellipsis
      data-something="something"
    >
      some text
    </Typography>
  ))).toMatchSnapshot();
});
