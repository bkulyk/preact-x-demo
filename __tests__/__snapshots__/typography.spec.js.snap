// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<span
  className="mdc-typography-base mdc-typography--body1"
>
  some text
</span>
`;

exports[`should render correctly with component prop 1`] = `
<p
  className="mdc-typography-base mdc-typography--headline5"
>
  some text
</p>
`;

exports[`should render correctly with many props 1`] = `
<div
  className="custom mdc-typography-base mdc-typography--overline mdc-typography-overflow-ellipsis"
  data-something="something"
>
  some text
</div>
`;
